lock_windows.go

```diff
@@ -0,0 +1,46 @@
+package dtest
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/datawire/dlib/dlog"
+	"golang.org/x/sys/windows"
```

Comment:
It's weird to me seeing `github.com/datawire` with 3rd-party imports.  We should set up the gci linter.